### ISSUE TITLE
Can't take BSQ swap offer through bisq daemon + bisq cli

### ISSUE THREAD

**Issue by pmartincalvo** [bisq-network/bisq#6353]
### Description

I am trying to sell my BTC to obtain BSQ by taking an existing offer, but the daemon insists that I don't have enough funds when I actually do.

Peculiar enough, if I try to sell BSQ to obtain BTC by taking an existng offer, the daemon notifies the offer has been taken and it appears that the trade has happened in the opposite direction.

The only way things I can imagine are happening are that something is swapped in the bisq cli, or I'm dumb and I'm missing something here.

Please, see the walkthrough below.

#### Version

I'm running version 1.9.4 of both daemon and cli.

### Steps to reproduce

---
#### Note
`bisq_cli` is an alias for `java -jar cli.jar --password=mypassword`

---

I have a BTC wallet with 0.01 BTC in it. To make sure of this, I run:
```bash
bisq_cli getbalance

BTC
Available Balance  Reserved Balance  Total Available Balance  Locked Balance
       0.01000000        0.00000000               0.01000000      0.00000000
BSQ
Available Confirmed Balance  Unverified Balance  Unconfirmed Change Balance  Locked For Voting Balance  Lockup Bonds Balance  Unlocking Bonds Balance
                       0.00                0.00                        0.00                       0.00                  0.00                     0.00
```

After this, I print the available offers like this:

```bash
bisq_cli getoffers --direction=SELL --currency-code=BSQ

Buy/Sell            Price in BTC for 1 BSQ           BTC(min - max)    BSQ(min - max)  Payment Method  Creation Date (UTC)   ID
Buy BSQ (Sell BTC)              0.00003820               0.00040000             10.47  BSQ Swap        2022-09-07T17:19:48Z  DRspF-8440a9ff-5f3a-4fb7-afcc-3a89203064e5-195   
Buy BSQ (Sell BTC)              0.00003801  0.00200000 - 0.00380000     52.62 - 99.97  BSQ Swap        2022-09-07T10:05:42Z  nolzwg-72aad6bb-1029-4a56-824c-a14749a21b43-195  
Buy BSQ (Sell BTC)              0.00003800  0.01000000 - 0.02000000   263.16 - 526.32  BSQ Swap        2022-09-05T02:16:20Z  8004691-fcbc828e-af6d-4a6f-9e0a-91d122e8a5d2-195 
Buy BSQ (Sell BTC)              0.00003740               0.00370000             98.93  BSQ Swap        2022-09-04T19:35:52Z  9207248-b346a2e7-bbb4-43df-a11e-bbbfb272881a-191 
Buy BSQ (Sell BTC)              0.00003720               0.01500000            403.23  BSQ Swap        2022-09-04T16:02:52Z  Egi2cwuK-7329fc96-172a-4b44-a87d-847341d2be48-195
Buy BSQ (Sell BTC)              0.00003701  0.00200000 - 0.03700000    54.04 - 999.73  BSQ Swap        2022-09-04T14:30:29Z  L8SQH-6c5b6b99-6523-4a88-95ff-802fb855b74c-195   
Buy BSQ (Sell BTC)              0.00003700  0.01000000 - 0.04000000  270.27 - 1081.08  BSQ Swap        2022-09-08T17:36:30Z  BI0BJS-243e5eb5-91e2-4655-8255-105fe31f3890-195  
Buy BSQ (Sell BTC)              0.00003700  0.00800000 - 0.01000000   216.22 - 270.27  BSQ Swap        2022-09-04T08:00:24Z  2FWLP-08a4c91e-7e52-44c4-9daa-e6be751350cc-195   
Buy BSQ (Sell BTC)              0.00003688  0.00200000 - 0.02000000    54.23 - 542.30  BSQ Swap        2022-08-30T12:00:03Z  69318-4df29d19-76ba-4bf1-a034-d3487aff0c6d-195   
Buy BSQ (Sell BTC)              0.00003600  0.02000000 - 0.08000000  555.56 - 2222.22  BSQ Swap        2022-09-05T02:17:37Z  8781275-85ee99d5-b9d4-4274-80d7-7b578a54cebe-195 
Buy BSQ (Sell BTC)              0.00003450  0.01000000 - 0.02960000   289.86 - 857.97  BSQ Swap        2022-07-20T16:30:43Z  553226-a4506858-4778-44df-b010-addfb1ee8e70-194  
Buy BSQ (Sell BTC)              0.00003440  0.01440000 - 0.02330000   418.60 - 677.33  BSQ Swap        2022-07-20T17:41:17Z  129672-82584ce9-4be6-4d4d-8d51-f16f5cc3a711-194  
Buy BSQ (Sell BTC)              0.00003300  0.00200000 - 0.00330000    60.61 - 100.00  BSQ Swap        2022-08-28T16:47:07Z  NGNRBHUE-067ad960-7e31-4928-9f48-8154e52655f7-194
Buy BSQ (Sell BTC)              0.00003285               0.00550000            167.43  BSQ Swap        2022-08-15T18:51:37Z  lr12ps-6245b833-502b-4bb8-b025-23f440ee8d54-194  
Buy BSQ (Sell BTC)              0.00002000               0.03500000           1750.00  BSQ Swap        2022-02-27T19:55:49Z  YHYGLVD-dd963c18-f2b2-44ba-8cac-f77dfa25da7a-182 
Buy BSQ (Sell BTC)              0.00001000               0.03000000           3000.00  BSQ Swap        2022-02-27T20:13:37Z  362264-05854b67-76d0-4f56-bd87-876e54b2f90b-182  
```

I decide I will take the first offer, `DRspF-8440a9ff-5f3a-4fb7-afcc-3a89203064e5-195`. I run as follows:

```bash
bisq_cli takeoffer --offer-id=DRspF-8440a9ff-5f3a-4fb7-afcc-3a89203064e5-195 --payment-account=BsqSwapAccount --fee-currency=BTC

Error: wallet has insufficient funds to take offer with id 'DRspF-8440a9ff-5f3a-4fb7-afcc-3a89203064e5-195'
```

This is the first unexpected result. I just checked that I have enough funds. Why does the command return this error?

Now is where things get even more strange. I'm going to check the offers again, but this time, the other way around:
```bash
bisq_cli getoffers --direction=BUY --currency-code=BSQ

Buy/Sell            Price in BTC for 1 BSQ           BTC(min - max)   BSQ(min - max)  Payment Method  Creation Date (UTC)   ID
Sell BSQ (Buy BTC)              0.00004000  0.00500000 - 0.02000000  125.00 - 500.00  BSQ Swap        2022-09-08T17:36:40Z  A3SCVTE-80e85357-7931-4b0f-a5b8-17b1da6d55e9-195 
Sell BSQ (Buy BTC)              0.00004200  0.01000000 - 0.04000000  238.10 - 952.38  BSQ Swap        2022-09-05T03:51:41Z  lnsqRNPb-60b2b20a-90bd-4e1c-8165-66d9ac78175e-194
Sell BSQ (Buy BTC)              0.00004321               0.04300000           995.14  BSQ Swap        2022-09-04T22:28:55Z  95d6OTH-96a93bb2-f51f-4175-889a-4c181528f773-195 
Sell BSQ (Buy BTC)              0.00004655               0.00070000            15.04  BSQ Swap        2022-07-20T20:53:08Z  asxly-427893e8-6d6b-4a12-97cc-3463cc217bdf-192   
Sell BSQ (Buy BTC)              0.00004679               0.00080000            17.10  BSQ Swap        2022-08-03T19:28:24Z  hgsZP-e7a80f22-7a06-416c-82aa-b4f5c8cbf798-192   
Sell BSQ (Buy BTC)              0.00004680               0.00060000            12.82  BSQ Swap        2022-07-20T20:49:31Z  PJNjEX-0461e540-c6ed-49f7-b1f7-8584f8aefb69-192  
Sell BSQ (Buy BTC)              0.00004724               0.00090000            19.05  BSQ Swap        2022-07-29T02:01:58Z  Fejwzd-1e0d10c9-8612-4001-b04e-fae90af42dc7-192  
Sell BSQ (Buy BTC)              0.00004800               0.04800000          1000.00  BSQ Swap        2022-09-06T22:53:03Z  73470897-d487ef8c-6586-43cf-8d43-fb49e9394392-195
Sell BSQ (Buy BTC)              0.00004800  0.00100000 - 0.00500000   20.83 - 104.17  BSQ Swap        2022-09-06T22:10:43Z  AEZZIY-d877fe14-e5a9-46d0-ae24-7df90f789040-194  
Sell BSQ (Buy BTC)              0.00004990               0.00050000            10.02  BSQ Swap        2022-07-29T02:02:54Z  pcokkua-06cdeb10-c423-4ef4-87f7-d711d4d23bf1-192 
Sell BSQ (Buy BTC)              0.00005000               0.00040000             8.00  BSQ Swap        2022-08-03T20:55:05Z  6ensso-15e41f4b-5444-4394-830d-199c8d2efc93-192  
Sell BSQ (Buy BTC)              0.00005100               0.05100000          1000.00  BSQ Swap        2022-09-04T14:52:29Z  EvWCGJT2-ec8961ee-405e-4104-8cfa-339209586208-195
Sell BSQ (Buy BTC)              0.00005499  0.00030000 - 0.00310000     5.46 - 56.37  BSQ Swap        2022-09-04T23:47:25Z  UZ9St-7109751f-81c1-4658-8ae1-94aa0a615545-195   
Sell BSQ (Buy BTC)              0.00005610               0.00040000             7.13  BSQ Swap        2022-07-29T01:59:34Z  3741221-a4290129-059b-47c6-a62b-e78e8613ecb9-192 
```

Theoretically, I can't take any of these offers, since my BSQ is balance is 0 and thus I have no BSQ I can sell. Regardless, I attempt to take offer `PJNjEX-0461e540-c6ed-49f7-b1f7-8584f8aefb69-192`:

```bash
bisq_cli takeoffer --offer-id=PJNjEX-0461e540-c6ed-49f7-b1f7-8584f8aefb69-192 --fee-currency=BTC

trade PJNjEX-0461e540-c6ed-49f7-b1f7-8584f8aefb69-192 successfully taken
```

And that is the second surprise. How was that taken if I don't have any BSQ I can sell?

Surely, if the offer had been taken, two side effects should have happened:
- The trade should appear in my trade history.
- My BTC and BSQ balances should have changed.

Let's check the balances first:
```bash
bisq_cli getbalance

BTC
Available Balance  Reserved Balance  Total Available Balance  Locked Balance
       0.00938658        0.00000000               0.00938658      0.00000000
BSQ
Available Confirmed Balance  Unverified Balance  Unconfirmed Change Balance  Locked For Voting Balance  Lockup Bonds Balance  Unlocking Bonds Balance
                       0.00               12.74                        0.00                       0.00                  0.00                     0.00
```

It seems the trade has happened, but I **bought** BSQ, not **sold** it.

Let's check my trade history:
```bash
bisq_cli gettrades --category=closed

Trade ID                                         Date/Time (UTC)       Market        Price  Deviation  Amount in BTC  Amount  Currency  Tx Fee (BTC)  Trade Fee  Buyer Deposit (BTC)  Seller Deposit (BTC)  Offer Type  Status
PJNjEX-0461e540-c6ed-49f7-b1f7-8584f8aefb69-192  2022-09-08T19:20:22Z  BSQ/BTC  0.00004680        N/A         0.0006   12.82  BSQ         0.00001342   0.00 BSQ           0.00000000            0.00000000  SELL BSQ    Pending
```

There is the trade. Clearly stating `SELL BSQ` as offer type.

Summing up: trades work in the opposite direction of what they announce.

### Expected behaviour

That trades where I buy BSQ and sell BTC work when I have 0.01 BTC and 0 BSQ.

That trades where I buy BTC and sell BSQ don't work when I have 0.01 BTC and 0 BSQ.

And that trades of `Offer Type` SELL BSQ decrease, not increase, my BSQ balance.

### Actual behaviour

The opposite of what is describe above.

### Screenshots

NA

#### Device or machine

Debian 11 on a VPS.

#### Additional info
NA

**Comment by ghubstan** [comment]
Gave me a fright(!), but it looks like it worked as it should.  The problem is the API reference not clearly stating what `Buy BSQ (Sell BTC)` means, and what `Sell BSQ (Buy BTC)` means.  See the `direction` field on https://bisq-network.github.io/slate/?java#offerinfo

The `Buy/Sell` direction is from the perspective the offer creator (maker), not yours (the taker).   Implementing what you describe in https://github.com/bisq-network/bisq/discussions/6332 would help avoid this confusion.

An offer to `Buy BSQ (Sell BTC)` is an offer to sell you BTC for BSQ.
The command `bisq_cli getoffers --direction=SELL --currency-code=BSQ` returns other peers' offers to sell BTC (to you, for BSQ).

An offer to `Sell BSQ (Buy BTC)` is an offer to buy BTC from you for BSQ.  
The command `bisq_cli getoffers --direction=BUY --currency-code=BSQ` returns other peers' offers to buy BTC (from you, with BSQ).

The first offer you tried to take `DRspF` is to sell you BTC for your BSQ `Buy BSQ (Sell BTC)`, and you had no BSQ to swap, hence the "wallet has insufficient funds" error.

You took the 2nd offer, `PJNjEX`, to sell *you* BSQ (Buy *your* BTC), and your Unverified BSQ Balance is 12.74.  If the tx has been confirmed, that 12.74 BSQ should now appear under `Available Confirmed Balance`.

**Comment by ghubstan** [comment]
All offers define the _offer makers'_ terms for buying or selling BTC, in exchange for some other currency.

**Comment by pmartincalvo** [comment]
Thanks a lot for shedding some light Stan. Glad to see it was only a misunderstanding and not really a bug. 

I agree that the way the direction is being shown might be a bit tricky. What I found most confusing, personally, was column `Buy/Sell` in the output of `getoffers`. Perhaps switching from `Sell BSQ (Buy BTC)` to something like `Maker sells BSQ (buys BTC)` would make things simpler for outsiders, although it's a bit bloaty.

Anyways, I'll leave it up to you. For now I'm happy understanding what was going on.
